js/functions.js

```diff
@@ -133,22 +130,230 @@ let trignoOperations = [
+  if (isNaN(value)) {
+    showErrForSomeTime();
+    return;
+  }
+  setCharAtInputField(value);
+}
+
+//to handle all the tan operations
+function tanTrignoOperations(clickedItem, value) {
+  switch (clickedItem) {
+    case "tan":
+      value = Math.tan(value);
+      break;
+    case "tan-in":
+      value = Math.atan(value);
+      break;
+    case "tan-h":
+      value = Math.tanh(value);
+      break;
+    case "tan-h-in":
+      if (value <= 1 && value >= -1) {
+        value = Math.atanh(value);
+      } else {
+        showErrForSomeTime("Please enter the input from -1 to 1 (RAD)");
+        return;
+      }
+      break;
+  }
+  if (isNaN(value)) {
+    showErrForSomeTime();
+    return;
+  }
+  setCharAtInputField(value);
+}
+
+//to handle all the cosec operation
+function cscTrignoOperations(clickedItem, value) {
+  switch (clickedItem) {
+    case "csc":
+      value = 1 / Math.sin(value);
+      break;
+    case "csc-in":
+      if (value <= 1 && value >= -1) {
+        value = 1 / Math.asin(value);
+        setCharAtInputField(value);
+      } else {
+        showErrForSomeTime("Please enter the input from -1 to 1 (RAD)");
+        return;
+      }
+    case "csc-h":
+      value = 1 / Math.sinh(value);
+      break;
+    case "csc-h-in":
+      value = 1 / Math.asinh(value);
+      break;
+  }
+  if (isNaN(value)) {
+    showErrForSomeTime();
+    return;
+  }
+  setCharAtInputField(value);
+}
+
+//to handle all the sec operations
+function secTrignoOperations(clickedItem, value) {
+  switch (clickedItem) {
+    case "sec":
+      value = 1 / Math.cos(value);
+      break;
+    case "sec-in":
+      if (value <= 1 && value >= -1) {
+        value = 1 / Math.acos(value);
+        setCharAtInputField(value);
+      } else {
+        showErrForSomeTime("Please enter the input from -1 to 1 (RAD)");
+        return;
+      }
+      break;
+    case "sec-h":
+      value = 1 / Math.cosh(value);
+      break;
+    case "sec-h-in":
+      value = 1 / Math.acosh(value);
+      break;
+  }
+  if (isNaN(value)) {
+    showErrForSomeTime();
+    return;
+  }
+  setCharAtInputField(value);
+}
+
+//to handle all the cot operations
+function cotTrignoOperations(clickedItem, value) {
+  switch (clickedItem) {
+    case "cot":
+      value = 1 / Math.tan(value);
+      break;
+    case "cot-in":
+      value = 1 / Math.atan(value);
+      break;
+    case "cot-h":
+      value = 1 / Math.tanh(value);
+      break;
+    case "cot-h-in":
+      if (value <= 1 && value >= -1) {
+        value = 1 / Math.atanh(value);
+      } else {
+        showErrForSomeTime("Please enter the input from -1 to 1 (RAD)");
+        return;
+      }
+      break;
+  }
+  if (isNaN(value)) {
+    showErrForSomeTime();
+    return;
+  }
+  setCharAtInputField(value);
 }
 
 //error handling function
 function showErrForSomeTime(string) {
-  document.getElementById("error-div").innerHTML = "Invalid Input!";
+  string !== undefined ? string : (string = "Invalid Input!");
```

Comment:
Please simplify this logic using `??`

Reply:
Also, we are querying `error-div` two times.
We can store the node as a constant in the function body, setTimeout will have access to the closure value.

Reply:
added default params and saved the ref of ele in variable